File: lab7/q1.py
import _thread
import time
import threading

	
class myThread (threading.Thread):
   def __init__(self, threadID, name, n):
      threading.Thread.__init__(self)
      self.threadID = threadID
      self.name = name
      self.n = n
      self.sum1 = 0
   def run(self):
   	  for i in range(1,self.n+1):
   	     if(self.n%i == 0):
   	        self.sum1 += i
	

print("Enter 2 numbers")
n1 = int(input())
n2 = int(input())
thread1 = myThread(1, "Thread-1", n1)
thread2 = myThread(2, "Thread-2", n2)
thread1.start()
thread2.start()
thread1.join()
thread2.join()
if(thread1.sum1 == thread2.sum1):
	print("Numbers are amicable")
else:
	print("Numbers are not amicable")

File: lab7/q1.java
import java.util.*;

class amicable extends Thread{
	int n,sum;
	public amicable(int n){
		this.n = n;
		this.sum = 0;
	}
	public void run(){
		for(int i=1;i<=n;i++){
			if(n%i == 0){
				sum += i;
			}
		}
	}
}

class q1{
	public static void main(String args[]){
		Scanner s = new Scanner(System.in);
		System.out.println("Enter 2 numbers");
		int n1 = s.nextInt();
		int n2 = s.nextInt();
		amicable t1 = new amicable(n1);
		amicable t2 = new amicable(n2);
		t1.start();
		t2.start();
		
		try{
			t1.join();
			t2.join();
		}
		catch(InterruptedException e){
			e.printStackTrace();
		}
		if(t1.sum == t2.sum){
			System.out.println("Numbers are amicable");
		}
		else{
			System.out.println("Number are not amicable");
		}
	
	}
	

}

